src/multivariate/mvlogitnormal.jl

```diff
@@ -0,0 +1,146 @@
+"""
+    MvLogitNormal
+
+The [multivariate logit-normal distribution](https://en.wikipedia.org/wiki/Logit-normal_distribution#Multivariate_generalization)
+is the multivariate generalization of [`LogitNormal`](@ref).
+
+It is a more general alternative to [`Dirichlet`](@ref) for probability vectors.
+
+If ``\\mathbf{y} \\sim \\mathrm{MvNormal}(\\boldsymbol{\\mu}, \\boldsymbol{\\Sigma})`` is a
+length ``d-1`` vector, then
+```math
+\\mathbf{x} = \\operatorname{softmax}\\left(\\begin{bmatrix}\\mathbf{y} \\\\ 0 \\end{bmatrix}\\right) \\sim \\mathrm{MvLogitNormal}(\\boldsymbol{\\mu}, \\boldsymbol{\\Sigma})
+```
+is a length ``d`` probability vector.
+
+```julia
+MvLogitNormal(μ, Σ)                 # MvLogitNormal with y ~ MvNormal(μ, Σ)
+MvLogitNormal(MvNormal(μ, Σ))       # same as above
+MvLogitNormal{MvNormal}(μ, Σ)       # same as above
+MvLogitNormal(MvNormalCanon(μ, J))  # MvLogitNormal with y ~ MvNormalCanon(μ, J)
+MvLogitNormal{MvNormalCanon}(μ, J)  # same as above
+```
+"""
+struct MvLogitNormal{D<:AbstractMvNormal} <: ContinuousMultivariateDistribution
+    normal::D
+    MvLogitNormal{D}(normal::D) where {D<:AbstractMvNormal} = new{D}(normal)
+end
+function MvLogitNormal{D}(args...) where {D<:AbstractMvNormal}
+    normal = D(args...)
+    return MvLogitNormal{typeof(normal)}(normal)
+end
+MvLogitNormal(d::AbstractMvNormal) = MvLogitNormal{typeof(d)}(d)
+MvLogitNormal(args...) = MvLogitNormal(MvNormal(args...))
+
+function Base.show(io::IO, d::MvLogitNormal)
```

Comment:
The generic methods treat the field names as parameters, which here looks weird because the field contains another distribution:
```julia
julia> using Distributions, LinearAlgebra

julia> d = MvLogitNormal(randn(3), exp(Symmetric(randn(3, 3))))
MvLogitNormal{FullNormal}(
dim: 3
μ: [-0.6695883087710813, -0.4262204488438936, 1.5508073295811842]
Σ: [1.8911987804702053 0.038814771826912765 1.2907448948413436; 0.038814771826912765 0.7054789545738211 0.06748072828787553; 1.2907448948413436 0.06748072828787553 0.9560938804684098]
)


julia> invoke(Base.show, Tuple{IO,Distribution}, stdout, d)
MvLogitNormal{FullNormal}(
normal: FullNormal(
dim: 3
μ: [-0.6695883087710813, -0.4262204488438936, 1.5508073295811842]
Σ: [1.8911987804702053 0.038814771826912765 1.2907448948413436; 0.038814771826912765 0.7054789545738211 0.06748072828787553; 1.2907448948413436 0.06748072828787553 0.9560938804684098]
)

)

julia> d = MvLogitNormal{MvNormalCanon}(randn(3), exp(Symmetric(randn(3, 3))))
MvLogitNormal{FullNormalCanon}(
μ: [-12.008515816447346, 9.221467028160104, 0.9303379007669893]
h: [-1.3761114231540177, 2.320365757512754, -0.05647292940406612]
J: [0.7714588053672484 0.9492070250024665 -0.929895944697952; 0.9492070250024665 1.6622992471750913 -1.7304512340037166; -0.929895944697952 -1.7304512340037166 5.088641347774431]
)


julia> invoke(Base.show, Tuple{IO,Distribution}, stdout, d)
MvLogitNormal{FullNormalCanon}(
normal: FullNormalCanon(
μ: [-12.008515816447346, 9.221467028160104, 0.9303379007669893]
h: [-1.3761114231540177, 2.320365757512754, -0.05647292940406612]
J: [0.7714588053672484 0.9492070250024665 -0.929895944697952; 0.9492070250024665 1.6622992471750913 -1.7304512340037166; -0.929895944697952 -1.7304512340037166 5.088641347774431]
)

)

```

We could use `show_multiline`, but this requires knowledge of the names of the parameters of the `AbstractMvNormal`, which we don't know, since Distributions doesn't have an API function for parameter names, just values.